mlir-sycl/test/Transforms/sycl-method-to-sycl-call.mlir

```diff
@@ -8,35 +8,85 @@
 !sycl_range_1_ = !sycl.range<[1], (!sycl.array<[1], (memref<1xi64, 4>)>)>
 !sycl_item_1_ = !sycl.item<[1, true], (!sycl.item_base<[1, true], (!sycl_range_1_, !sycl_id_1_, !sycl_id_1_)>)>
 
-// CHECK-LABEL: func.func @accessor_subscript_operator(%arg0: memref<?x!sycl_accessor_2_i32_rw_gb, 4>, %arg1: memref<?x!sycl_id_2_>) -> memref<?xi32, 4> {
-// CHECK-NEXT: %0 = sycl.call(%arg0, %arg1) {FunctionName = @"operator[]", MangledFunctionName = @_ZNK4sycl3_V18accessorIiLi2ELNS0_6access4modeE1026ELNS2_6targetE2014ELNS2_11placeholderE0ENS0_3ext6oneapi22accessor_property_listIJEEEEixILi2EvEERiNS0_2idILi2EEE, TypeName = @accessor} : (memref<?x!sycl_accessor_2_i32_rw_gb, 4>, memref<?x!sycl_id_2_>) -> memref<?xi32, 4>
+// CHECK-LABEL:   func.func @accessor_subscript_operator(
+// CHECK-SAME:                                           %[[VAL_0:.*]]: !sycl_accessor_2_i32_rw_gb,
+// CHECK-SAME:                                           %[[VAL_1:.*]]: memref<?x!sycl_id_2_>) -> memref<?xi32, 4> {
+// CHECK-NEXT:      %[[VAL_2:.*]] = arith.constant 0 : index
+// CHECK-NEXT:      %[[VAL_3:.*]] = memref.alloca() : memref<1x!sycl_accessor_2_i32_rw_gb>
+// CHECK-NEXT:      %[[VAL_4:.*]] = memref.alloca() : memref<1xindex>
+// CHECK-NEXT:      memref.store %[[VAL_0]], %[[VAL_3]]{{\[}}%[[VAL_2]]] : memref<1x!sycl_accessor_2_i32_rw_gb>
+// CHECK-NEXT:      memref.store %[[VAL_2]], %[[VAL_4]]{{\[}}%[[VAL_2]]] : memref<1xindex>
+// CHECK-NEXT:      %[[VAL_5:.*]] = memref.reshape %[[VAL_3]](%[[VAL_4]]) : (memref<1x!sycl_accessor_2_i32_rw_gb>, memref<1xindex>) -> memref<0x!sycl_accessor_2_i32_rw_gb>
```

Comment:
This is an illegal reshape operation because the result has zero element while the operand has 1 element. The specification (https://mlir.llvm.org/docs/Dialects/MemRef/#memrefreshape-mlirmemrefreshapeop) states:
"The source and destination types are compatible if both have the same element type, same number of elements, address space and identity layout map".

Reply:
I usually see that we use `memref.cast`, is there any reasons why we want to use `memref.reshape` in this case?
Thinking that having one consistent way could hope writing optimizations.

Reply:
Looking at https://mlir.llvm.org/docs/Dialects/MemRef/#memrefcast-mlirmemrefcastop examples, our use case here seems to match perfectly with one of their examples: `Erase static shape information, replacing it with dynamic information.`, so I think `memref.cast` is preferable here.

Reply:
My bad: forgot to update tests. Done!

Reply:
Yes using memref.cast is what we need here.